### PR TITLE
blog: inflight magazine #2

### PR DESCRIPTION
**Pull request by revitalbarletz**
Resolves https://github.com/monadahq/marketing/issues/37

### CODE REVIEW

File: blog/2023-3-5-magazine-002.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+Thank you 🙏 for your support and feedback, and please keep it coming! We're always looking for ways to improve and make Wing the best it can be.
+
+## TL;DR
+
+You received this email because we have you on our list as someone who might be interested to stay informed about the Wing Programming Language.
+Here is a summary of the topics in today's issue:
+ - 🔍 [Check out the latest Wingly Update episode](#check-out-the-latest-wingly-update-episode) 
+ - 🔍 [Wing User Stories are now published to our doc site](#wing-user-stories-are-now-published-to-our-doc-site)
+ - 🔍 [Wing Console​ gets a new map view](#wing-console-gets-a-new-map-view) - New look and feel for the wing console application view based on ELK algorithm
+ - 🔍 Deep dive: user resources  - Create reusable building blocks organized into logical units that abstract both functional logic and infrastructure
```
Comment (eladb) [inline blog/2023-3-5-magazine-002.md:36]: missing links here

File: blog/2023-3-5-magazine-002.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+the [`cloud.Api`](https://github.com/winglang/wing/issues/623) resource, 
+a [`Redis` resource](https://github.com/winglang/wing/issues/611)
+and [interface declarations]((https://github.com/winglang/wing/issues/123)).
+
+### Wing Console gets a new map view
+
+The [Wing Console](https://docs.winglang.io/getting-started/console) is a desktop application that can be used to interact with Wing applications
+and its resources on the local machine. 
+One of the biggest challenges of the console is how to visualize a complex real-world cloud applications. In the latest release, we've leveraged the [ELK layout algorithm](https://www.eclipse.org/elk/documentation.html) to create an interactive map of the resources generated by Wing so and visualize the resource hierarchy and their connections.
+
```
Comment (eladb) [inline blog/2023-3-5-magazine-002.md:73]: Missing screenshots

File: blog/2023-3-5-magazine-002.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  allowfullscreen>
+</iframe>
+  
+### Compiler errors and the "blue screen of death"
+
+When the console compiles your Wing code in the background, if it encounters compiler errors, we now display them in a view that is a homage to the Windows [Blue Screen of Death](https://en.wikipedia.org/wiki/Blue_screen_of_death). File locations are links and will jump right into your IDE.
+
+Here's a short video:
+
+<iframe
```
Comment (eladb) [inline blog/2023-3-5-magazine-002.md:98]: This doesn't render well

**Review by eladb — CHANGES_REQUESTED** [review]
I pushed a [commit] with some updates.

Left a few more comments!

[commit]: https://github.com/winglang/docsite/pull/313/commits/ee676b4dd8cabaa89e77f442209f2b9fab7b562d

File: blog/2023-3-5-magazine-002.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+We have improved its view to a map view! The idea is to use [elkjs](https://github.com/kieler/elkjs) to render an interactive map of the resources generated by Wing so that we can visualize the resource hierarchy and their connections easily. The map view uses the ELK algorithm, which generates a graph of your cloud resources. [The ELK algorithm](https://www.eclipse.org/elk/documentation.html) is a layout engine that can deal with both a hierarchical structure (the resource tree) and an interaction model (the relationships between resources).
+
+With this new map view, you can visualize both the static structure and the data flow of your cloud application at a glance. Cloud resources are connected to each other depending on usage, giving you a clear view of the dependencies and interactions between different parts of your application. You can quickly identify any bottlenecks or issues that may be affecting the performance of your application.
+
+#### Wing Console adds support for filtering logs
+
+We've also added a log filtering feature to help you easily filter logs by word or level! 
+Check out this video for more details and usage:
+
+<iframe
```
Comment (revitalbarletz) [inline blog/2023-3-5-magazine-002.md:82]: Fix

File: blog/2023-3-5-magazine-002.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" 
+  allowfullscreen>
+</iframe>
+  
+#### Homage to the “blue screen of death”
+
+And, for your convenience, we've introduced a blue screen of death to show compiler error, making it easier to restore and navigate to the error location… very nostalgic! Hopefully it doesn't trigger anyone ;) 
+And a short video that shows this feature can be found here: 
+
+<iframe
```
Comment (revitalbarletz) [inline blog/2023-3-5-magazine-002.md:98]: fix

File: blog/2023-3-5-magazine-002.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+## TL;DR
+
+You received this email because we have you on our list as someone who might be interested to stay informed about the Wing Programming Language.
+Here is a summary of the topics in today's issue:
+ - 🔍 [Check out the latest Wingly Update episode](#check-out-the-latest-wingly-update-episode) 
+ - 🔍 [Wing User Stories are now published to our doc site](#wing-user-stories-are-now-published-to-our-doc-site)
+ - 🔍 [Wing Console​ gets a new map view](#wing-console-gets-a-new-map-view) - New look and feel for the wing console application view based on ELK algorithm
+ - 🔍 Deep dive: user resources  - Create reusable building blocks organized into logical units that abstract both functional logic and infrastructure
+ - 🚀 New: Introducing duration and datetime types - our primary engineering focus is to finish implementing the compiler so we can actually start using Wing to build some real stuff
```
Comment (revitalbarletz) [inline blog/2023-3-5-magazine-002.md:37]: Add links to the ones below

File: blog/2023-3-5-magazine-002.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+We are thrilled to see pull requests keep popping up from:
+
+[@Kasey Abu-Rayyan](https://github.com/winglang/wing/pull/1431), 
+[@Marcio Cruz](https://github.com/winglang/wing/pull/1663), 
+[@Raphael Manke](https://github.com/winglang/wing/pull/1220), 
+[@Eric Johnson](https://github.com/winglang/wing/pull/1009),  
+[@Joshua Dando](https://github.com/winglang/wing/pull/1018), 
+[@Yariv Levi](https://github.com/winglang/wing/pull/1692), 
+[@Ananthu C V](https://github.com/winglang/wing/pull/1137), 
+[@Guy Waldeman](https://github.com/winglang/wing/issues/1123),
```
Comment (staycoolcall911) [inline blog/2023-3-5-magazine-002.md:228]: Let's remove Guy for now - this issue is not done and not assigned to him.
```suggestion
```

File: blog/2023-3-5-magazine-002.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+ - 🔍 [Check out the latest Wingly Update episode](#check-out-the-latest-wingly-update-episode) 
+ - 🔍 [Wing User Stories are now published to our doc site](#wing-user-stories-are-now-published-to-our-doc-site)
+ - 🔍 [Wing Console​ gets a new map view](#wing-console-gets-a-new-map-view) - New look and feel for the wing console application view based on ELK algorithm
+ - 🔍 Deep dive: user resources  - Create reusable building blocks organized into logical units that abstract both functional logic and infrastructure
+ - 🚀 New: Introducing duration and datetime types - our primary engineering focus is to finish implementing the compiler so we can actually start using Wing to build some real stuff
+ - 🚀 New: json type support  - First class support for JSON in Wing
+ - 🚀 New: copy and copy_mut  - Convert mutable and immutable types of wing primitive collections
+ - 🗣️ Updates to our contribution guide - Easier navigation through the guide and accessibility to  day-to-day workflows
+
```
Comment (staycoolcall911) [inline blog/2023-3-5-magazine-002.md:41]: Can we share that we are hiring, please?
https://www.linkedin.com/jobs/view/3482716147/

Comment (eladb) [reply blog/2023-3-5-magazine-002.md:41]: Yes, also doc writer!

File: blog/2023-3-5-magazine-002.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+  allowfullscreen>
+</iframe>
+
+### Introducing new types to the language 
+
+We are still enriching the [Wing Language Specification](https://docs.winglang.io/reference/spec) and increasing the set of features we believe are required to build cloud applications with Wing.
+
+We added support for `for..in` ([#1412](https://github.com/winglang/wing/pull/1412/)) as well as the `duration` and `datetime` types ([#1470](https://github.com/winglang/wing/pull/1470)).
+
+The `duration` and the `datetime` type are also [built-in](https://docs.winglang.io/reference/spec#116-datetime) types for Wing now. We have literals for minutes, seconds, and hours. We also have an initial API for `datetime` objects.
```
Comment (eladb) [inline blog/2023-3-5-magazine-002.md:183]:
```suggestion
We added support for `for..in` ([#1412](https://github.com/winglang/wing/pull/1412/)) as well as the `duration` and `datetime` types ([#1470](https://github.com/winglang/wing/pull/1470)). There are also duration literals minutes (e.g `5m`), seconds (e.g. `10s`), and hours (e.g. `5h`).
```

**Review by eladb — CHANGES_REQUESTED** [review]

File: blog/2023-3-5-magazine-002.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+## TS;GMM (too short; give me more)
+
+### Check out the latest Wingly Update episode
+
+**The Wingly Update** is a live show we are streaming on Twitch. We mostly make fools of ourselves, talk about the latest developments of the project, bring guests and more.
+
+In our last show, [Eyal](https://www.linkedin.com/in/eyal-keren-b23b7231/) hosted [Elad](https://www.linkedin.com/in/hackingonstuff),
+[Cristian](https://www.linkedin.com/in/cristian-pallarés-88139916/), [Mark the DJ](https://www.linkedin.com/in/mark-mcculloh/).
+
+We talked about the new map view in the Wing Console, user-defined resources and played some music.
```
[the remark this answers is not in the document]
Comment (revitalbarletz) [reply blog/2023-3-5-magazine-002.md:55]: This is no longer the last show, do we want to wait for the last show to be uploaded?

[the remark this answers is not in the document]
Comment (revitalbarletz) [reply blog/2023-3-5-magazine-002.md]: Done.

File: blog/2023-3-5-magazine-002.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+  allowfullscreen>
+</iframe>
+
+### Introducing new types to the language 
+
+We are still enriching the [Wing Language Specification](https://docs.winglang.io/reference/spec) and increasing the set of features we believe are required to build cloud applications with Wing.
+
+We added support for `for..in` ([#1412](https://github.com/winglang/wing/pull/1412/)) as well as the `duration` and `datetime` types ([#1470](https://github.com/winglang/wing/pull/1470)). There are also duration literals minutes (e.g `5m`), seconds (e.g. `10s`), and hours (e.g. `5h`).
+
+Let's take a look at a quick example:
```
Comment (eladb) [inline blog/2023-3-5-magazine-002.md:188]: Maybe also add an example of the `duration` type:

```js
let timeout = 5m;
assert(timeout.seconds == 300);
```

**Review by eladb — COMMENTED** [review]
One more comment

File: blog/2023-3-5-magazine-002.md (hunk 11 above)
Comment (revitalbarletz) [reply blog/2023-3-5-magazine-002.md:188]: Done.

**Pull request merged**